homeassistant/components/vallox/button.py

```diff
@@ -0,0 +1,80 @@
+"""Support for Vallox buttons."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import date
+
+from vallox_websocket_api import Vallox
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ValloxDataUpdateCoordinator, ValloxEntity
+from .const import DOMAIN
+
+
+class ValloxButtonEntity(ValloxEntity, ButtonEntity):
+    """Representation of a Vallox button."""
+
+    entity_description: ValloxButtonEntityDescription
+    _attr_entity_category = EntityCategory.CONFIG
+
+    def __init__(
+        self,
+        name: str,
+        coordinator: ValloxDataUpdateCoordinator,
+        description: ValloxButtonEntityDescription,
+        client: Vallox,
+    ) -> None:
+        """Initialize the Vallox button."""
+        super().__init__(name, coordinator)
+
+        self.entity_description = description
+
+        self._attr_unique_id = f"{self._device_uuid}-{description.key}"
+        self._client = client
+
+
+class ValloxResetFilterChangeDateButton(ValloxButtonEntity):
+    """Representation of a Vallox reset filter button."""
+
+    async def async_press(self) -> None:
+        """Press the button."""
+        await self._client.set_filter_change_date(date.today())
+        await self.coordinator.async_request_refresh()
```

Comment:
Instead of putting this here, you can extend the ValloxButtonEntityDescription with a press_fn and put the function in there. Check Flexit_bacnet (I think the number entities) for an example

Reply:
I was thinking `ValloxButtonEntityDescription` will be stored somewhere, so it would be bad to extend it with non serializable data. Will take a look.

Reply:
I do not think this integration would ever have more than one DateEntity. Filter change date is the only only one that unit exposes.

Reply:
But then we shouldn't use the entity description. Although this works, this isn't the optimal way of using either method

Reply:
I see DateEntity class has entity_description: DateEntityDescription parameter. No idea what it is used for.

Reply:
Well the idea behind an entity description is a way to quickly create new entities. For example when you look at a `device_class` property:

```py
    def device_class(self) -> SensorDeviceClass | None:
        """Return the class of this entity."""
        if hasattr(self, "_attr_device_class"):
            return self._attr_device_class
        if hasattr(self, "entity_description"):
            return self.entity_description.device_class
        return None
```

So instead of having 50 classes with each their own variation, we have an entity description, a class we can inject and then the trick is to encapsulate all the variation in that entity description. Your `SensorEntity` (as example) would then be a generic class that gets the right values from the entity descriptions to still express that variance.